File: NumberGuessingGame.py
# to import random module
import random
# to create a range of random numbers between 1-10
n = random.randrange(1,100)
# to take a user input to enter a number
guess = int(input("Enter any number: "))
while n!= guess: # means if n is not equal to the input guess
    # if guess is smaller than n
    if guess < n:
        print("Too low")
        # to again ask for input
        guess = int(input("Enter number again: "))
    # if guess is greater than n
    elif guess > n:
        print("Too high!")
        # to again ask for the user input
        guess = int(input("Enter number again: "))
    # if guess gets equals to n terminate the while loop
    else:
        break
print("you guessed it right!!")



=============================================================
========================= OR ================================
=============================================================

import random

secret_number = random.randint(1, 100)
attempts = 0

print("🎯 Number Guessing Game")
print("Guess a number between 1 and 100")

while True:
    guess = int(input("Enter your guess: "))
    attempts += 1

    if guess < secret_number:
        print("📉 Too low!")
    elif guess > secret_number:
        print("📈 Too high!")
    else:
        print(f"🎉 Correct! You guessed it in {attempts} attempts.")
        break

File: NumberGuessingGame.java
import java.util.Random;
import java.util.Scanner;

public class NumberGuessingGame {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        Random random = new Random();

        int number = random.nextInt(100) + 1;
        int guess;
        int attempts = 0;

        System.out.println("===== Number Guessing Game =====");
        System.out.println("I have selected a number between 1 and 100.");
        System.out.println("Try to guess it!");

        while (true) {

            System.out.print("Enter your guess: ");
            guess = sc.nextInt();

            attempts++;

            if (guess < number) {
                System.out.println("Too Low! Try again.");
            }
            else if (guess > number) {
                System.out.println("Too High! Try again.");
            }
            else {
                System.out.println("🎉 Congratulations!");
                System.out.println("You guessed the correct number.");
                System.out.println("Number of attempts: " + attempts);
                break;
            }
        }

        sc.close();
    }
}
